src/mscorlib/src/System/Version.cs

```diff
@@ -29,46 +29,25 @@ public sealed class Version : ICloneable, IComparable
         , IComparable<Version>, IEquatable<Version>
     {
         // AssemblyName depends on the order staying the same
-        private int _Major;
-        private int _Minor;
-        private int _Build = -1;
-        private int _Revision = -1;
+        private readonly int _Major;
+        private readonly int _Minor;
+        private readonly int _Build = -1;
+        private readonly int _Revision = -1;
         private static readonly char[] SeparatorsArray = new char[] { '.' };
     
-        public Version(int major, int minor, int build, int revision) {
-            if (major < 0) 
-              throw new ArgumentOutOfRangeException("major",Environment.GetResourceString("ArgumentOutOfRange_Version"));
-
-            if (minor < 0) 
-              throw new ArgumentOutOfRangeException("minor",Environment.GetResourceString("ArgumentOutOfRange_Version"));
-
-            if (build < 0)
-              throw new ArgumentOutOfRangeException("build",Environment.GetResourceString("ArgumentOutOfRange_Version"));
-            
-            if (revision < 0) 
-              throw new ArgumentOutOfRangeException("revision",Environment.GetResourceString("ArgumentOutOfRange_Version"));
+        public Version(int major, int minor, int build, int revision) : this(major, minor, build) {
+            if (revision < 0)
```

Comment:
This looks like one trade-off between speed and size ... I would expect that the Version(int,int,int,int) constructor will be measurably slower with this change.